### ISSUE TITLE
Sort doesn't sort int values

### ISSUE THREAD

**Issue by lvictorino** [Kinto/kinto#534]
These records are extracted from my PGSQL 9.5 database :
```
{"id": "5806c7b7-ee11-46af-b36b-f7d03be0aabb", "hash": "FE1B4FD1BE3C98DC277D702A59DE6942", "time": "1459866369", "score": 1, "platform": "WindowsEditor", "total_time": 8.473255}
{"id": "f1043d04-e920-4615-9cb4-d058a48b5c8e", "hash": "9F41FBDC2B04AEA8D5B283963A60BBB1", "time": "1459866454", "score": 4, "platform": "WindowsEditor", "total_time": 12.5855}
{"id": "aa83a166-490e-4797-b056-f06f5d23611e", "hash": "8DDF97D15D65EB94651CC2C46B56E79C", "time": "1459866564", "score": 4, "platform": "WindowsEditor", "total_time": 7.508384}
{"id": "880e2e8d-b4c2-445e-9bf5-15b73959a2e7", "hash": "B6746C8D6A265DD55B48EF0607BCE95A", "time": "1459866606", "score": 6, "platform": "WindowsEditor", "total_time": 15.91712}
{"id": "77431dde-674a-460d-a0cd-a000e0603da2", "hash": "BD1AA97D81112DE197BDB569257BE110", "time": "1459866686", "score": 41, "platform": "WindowsEditor", "total_time": 56.66311}
{"id": "c43f72b0-f683-4871-82b4-4b6353a0d6de", "hash": "D6C419455424E10C8BF6DBDEDC25585C", "time": "1459866768", "score": 3, "platform": "WindowsEditor", "total_time": 11.63698}
````
When I make a request on these records with a `_sort` parameter ( /v1/buckets/default/collections/bar/records?_sort=-score ) the first returned score is 6 and not  41. 

It seems that the scores are sorted like if they were strings and not integers (6 > 4).

**Comment by Natim** [comment]
This appear to be the case with the Postgresql backend but not with the Memory backend.

**Comment by magopian** [comment]
Yes, just reproduced with postgresql9.4 (but not with the in-memory backend)

**Comment by Natim** [comment]
> Yes, just reproduced with postgresql9.4 (but not with the in-memory backend)

We shall add a storage test :+1:

**Comment by magopian** [comment]
Here's the SQL query we're sending to posgresql:
```
SELECT total_filtered.count AS count_total,
               a.id, as_epoch(a.last_modified) AS last_modified, a.data
          FROM paginated_records AS p JOIN all_records AS a ON (a.id = p.id),
               total_filtered
          ORDER BY data->>(:sort_field_0) DESC, last_modified DESC;
```

And here are the parameters:
```
         {'sort_field_0': u'score', 'collection_id': 'record', 'deleted_field': '{"deleted":true}', 'parent_id': u'/buckets/4ed2d0f8-ba97-9765-af32-d19b001f70d8/collections/foobar'}
```

If we try running the following query straight in psql, we can confirm that the order is coherent with a string sort, not an integer sort.

**Comment by magopian** [comment]
Ok, @Natim found the answer: we should be using `data->` (to json value), instead of `data->>` (to text).

**Comment by almet** [comment]
Hi @lvictorino and thanks for opening this bug.

I'm surprised this wasn't tested. While chatting with @natim yesterday, he told me that the storage tests aren't really run on each backend. If this is true, we might want to run all the storage tests for each backend. @Natim, can you confirm, and eventually open a bug about this? Thanks!

**Comment by Natim** [comment]
No all the storage tests are run with all backends, but we do not run all cliquet tests with all backends.

**Comment by almet** [comment]
> No all the storage tests are run with all backends, but we do not run all cliquet tests with all backends.

I don't understand what you mean. Can you rephrase?

**Comment by Natim** [comment]
> he told me that the storage tests aren't really run on each backend

This is not what I said. If you look at ``test_storage.py`` you can see that they are run for each backend.

This bug wasn't covered by the storage tests.

What I said is that cliquet tests are [mostly run with the Redis backend](https://github.com/mozilla-services/cliquet/blob/master/cliquet/tests/support.py#L101-L103).

It could be a great idea ot add one running with Postgresql in the travis matrix.

**Comment by Natim** [comment]
This have been fixed in Cliquet 3.1.1
